libnetdata/queue/queue.h

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#ifndef NETDATA_QUEUE_H
+#define NETDATA_QUEUE_H 
+
+#include "../libnetdata.h"
+		
+typedef struct nodes				
+{
+    void *item;
+    struct nodes *next;			
+} node;
+
+typedef struct Queues			
+{		
+    pthread_cond_t conde;
+    pthread_cond_t condf;
+    pthread_mutex_t lock;
+    node *front;			
+    node *rear;
+    int max;
+    int count;
+} Queue;
```

Comment:
Don't expose the internal implementation of these structs. Provide an opaque pointer instead.

Reply:
Done.